contracts/Depository.sol

```diff
@@ -226,56 +234,55 @@ contract Depository is IErrorsTokenomics {
 
         // Push newly created bond product into the list of products
         productId = productCounter;
-        mapBondProducts[productId] = Product(uint160(priceLP), uint32(expiry), token, uint96(supply));
+        mapBondProducts[productId] = Product(uint160(priceLP), uint32(vesting), token, uint96(supply));
         // Even if we create a bond product every second, 2^32 - 1 is enough for the next 136 years
         productCounter = uint32(productId + 1);
-        emit CreateProduct(token, productId, supply, priceLP, expiry);
+        emit CreateProduct(token, productId, supply, priceLP, vesting, block.timestamp);
     }
 
     /// @dev Closes bonding products.
     /// @notice This will terminate programs regardless of their vesting time.
     /// @param productIds Set of product Ids.
+    /// @return numClosedProducts Number of closed products.
     /// #if_succeeds {:msg "productCounter not touched"} productCounter == old(productCounter);
```

Comment:
Add a fuzzing test to show that product being closed still allows calling this

Reply:
Done in foundry.